test/unified_format.py

```diff
@@ -1011,7 +1024,7 @@ def run_entity_operation(self, spec):
                 return self.process_error(exc, expect_error)
             raise
         else:
-            if expect_error:
+            if expect_error and opname != "iterateUntilDocumentOrError":
```

Comment:
What is the iterateUntilDocumentOrError check for?

Reply:
For the case where this an "expectError", but the error condition isn't raised, because we got a document instead, which as I understand it is what `iterateUntilDocumentOrError` implies.

Reply:
It looks like iterateUntilDocumentOrError is usually (always?) accompanied by "expectError" or "expectResult". When expectError is there but the iterateUntilDocumentOrError op does not error, shouldn't we still fail the test?

Reply:
I removed the modification based on feedback about how "expectError" should be handled in the spec: "If [operation.expectError](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#operation-expecterror) is specified, the test runner MUST assert that the operation yielded an error; otherwise, the test runner MUST assert that the operation did not yield an error.

I suspect this will make "test_change_stream.TestUnifiedChangeStreamsErrors.test_change_stream_errors_on_ElectionInProgress" fail again on Server 4.2, let's find out.

Reply:
Sure enough:

```python
AssertionError: Excepted error {'errorCode': 216} but "iterateUntilDocumentOrError" succeeded: 
{'_id':
 {'_data': '82628D50B2000000042B022C0100296E5A10044BBFDAAD009A4B0A8CF9E843E82603C946645F69640064628D50B26500615C7FE84A840004'},
 'operationType': 'insert', 'clusterTime': Timestamp(1653428402, 4), 'fullDocument':
 {'_id': ObjectId('628d50b26500615c7fe84a84'), 'z': 3}, 'ns': {'db': 'database0', 'coll': 'collection0'},
 'documentKey': {'_id': ObjectId('628d50b26500615c7fe84a84')}}
```

Reply:
I don't think it is a spec test failure, given that several other drivers have implemented the unified spec change.

Reply:
How should we proceed?

Reply:
I'll do some testing with server 4.2 to see if I can spot a bug in our driver.